worker/caasprober/controller.go

```diff
@@ -47,30 +49,30 @@ func (c *Controller) Kill() {
 }
 
 func (c *Controller) makeLoop(
-	probes CAASProbes,
+	probes *CAASProbes,
 	mux Mux,
 ) func() error {
 	return func() error {
 		if err := mux.AddHandler(
 			http.MethodGet,
 			k8sconstants.AgentHTTPPathLiveness,
-			ProbeHandler("liveness", probes.LivenessProbe)); err != nil {
+			ProbeHandler("liveness", probes.Liveness)); err != nil {
```

Comment:
we could replace this with (slightly pseudo)
`ProbeHandler("liveness", probes.getProbe("liveness))` (returns a noop prober if not supported)

and use a loop for the AddHandler calls